sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -322,6 +327,14 @@ private URI getUri(RxDocumentServiceRequest request) throws URISyntaxException {
             null);
     }
 
+    private boolean isEmulatorHost(URI uri) {
+        if (StringUtils.isNotEmpty(uri.getHost())) {
+            return uri.getHost().equals("localhost") || uri.getHost().equals("127.0.0.1");
```

Comment:
Isn't it a bit artificial? Why only support ipv4 and not ipv6? I also don't quite get why you would want to suppress emulator usage when the emulator is hosted publicly. Woudln't it make more sense to validate let's say responses with just http are expected to have a certain header - indicating that this is an emulator endpoint?

Reply:
Good point, I could just change to allow use http for any endpoints, but also want to avoid the situation public endpoint is being used accidently.